1.regression_modelling/scripts/0.split_data.py

```diff
@@ -0,0 +1,139 @@
+).resolve(strict=True)
+
+# load all normalized parquet files
+files = norm_path.glob("*_sc_normalized.parquet")
+dfs = [pd.read_parquet(file) for file in files]
+
+# concatenate them into one data frame
+combined_df = pd.concat(dfs, ignore_index=True)
+
+# perform feature selection to drop any columns that have NaN (avoid downstream issues)
+combined_df = feature_select(
+    combined_df,
+    operation="drop_na_columns",
+    na_cutoff=0
+)
+
+# print df
+print(combined_df.shape)
+combined_df.head()
+
+
+# ## Load in filtered single-cell data from first module
+
+# In[3]:
+
+
+# load in filtered tuple dataframe
+filtering_df = pd.read_parquet(
+    pathlib.Path(
+        "../0.data_analysis_and_processing/filtered_single_cells/filtered_single_cell_profiles.parquet"
+    )
+)
+
+# print df
+print(filtering_df.shape)
+filtering_df.head()
+
+
+# ## Filter out any cells that do not matched the filtered data tuple
+
+# In[4]:
+
+
+# define the columns to match that are important for filtering
+columns_to_match = [
+    'Metadata_Plate', 'Metadata_Well', 'Metadata_Site',
+    'Metadata_Nuclei_Location_Center_X', 'Metadata_Nuclei_Location_Center_Y'
+]
+
+# ensure both dataframes have these columns
+combined_df_filtered = combined_df[columns_to_match]
+filtering_df_filtered = filtering_df[columns_to_match]
+
+# convert rows to tuples
+filtering_tuples = set(tuple(row) for row in filtering_df_filtered.itertuples(index=False))
+combined_df_tuples = combined_df_filtered.apply(tuple, axis=1)
+
+# create a boolean mask for rows in combined_df that match any rows in filtering_df
+mask = combined_df_tuples.isin(filtering_tuples)
+
+# filter combined_df to keep only rows that match any row in filtering_df
+filtered_combined_df = combined_df[mask]
+
+# reset index
+filtered_combined_df.reset_index(drop=True, inplace=True)
+
+print(filtered_combined_df.shape)
+filtered_combined_df.head()
+
+
+# ## Hold out all untreated single-cells from the `293T` cell line
+
+# In[5]:
+
+
+# set path for training, testing, and holdout datasets
+data_dir = pathlib.Path("./data")
+data_dir.mkdir(exist_ok=True)
+
+# holdout all cells from the 293T cell line as CSV
+holdout_data = filtered_combined_df[filtered_combined_df['Metadata_CellLine'] == '293T']
+
+# Save the holdout data to a CSV file
+holdout_data.to_csv(f"{data_dir}/holdout_data_293T.csv", index=False)
+
+# Print the shape of holdout_data to verify
+print(holdout_data.shape)
+
+
+# ## Split data 70% training and 30% testing
+
+# In[6]:
+
+
+# Remove rows with Metadata_CellLine as 293T
+remaining_data = filtered_combined_df[filtered_combined_df['Metadata_CellLine'] != '293T']
+
+# Split the data into training and testing sets
+train_data, test_data = train_test_split(
+    remaining_data,
+    test_size=0.3,
+    random_state=0,  # For reproducibility
+    shuffle=True     # Ensure data is shuffled before splitting
+)
+
```

Comment:
While doing your data splits, are there any variables that the data should be stratified over? 

If there are multiple cell types it might be worth ensuring that each cell type has the same ratio of representation in each split. Same for treatment (drug or RNAi or CRISPRi or ORF, etc) 

You know the data best, so I will leave this up to you to decide.

Reply:
Fair points!

There are only siRNA-treated cells left after removing holdout from one cell line. We don't have any real reason to stratify these treatments in a specific way, so for now this is pretty good enough. Thank you for bringing up these points!

Reply:
Awesome, thank you for clarifying!